edgehog-device-runtime-containers/src/service/mod.rs

```diff
@@ -416,61 +381,82 @@ impl<D> Service<D> {
     where
         D: Client + Sync + 'static,
     {
-        let id = Id::new(ResourceType::Deployment, id);
-        debug!("stopping {id}");
+        let containers = match self.store.load_deployment_containers(id).await {
+            Ok(Some(containers)) => containers,
+            Ok(None) => {
+                error!("{id} not found");
 
-        let Some(node) = self.nodes.node(&id) else {
-            error!("{id} not found");
+                DeploymentEvent::new(EventStatus::Error, format!("{id} not found"))
+                    .send(&id, &self.device)
+                    .await;
 
-            DeploymentEvent::new(EventStatus::Error, format!("{id} not found"))
-                .send(id.uuid(), &self.device)
-                .await;
+                return;
+            }
+            Err(err) => {
+                let err = format!("{:#}", eyre::Report::new(err));
 
-            return;
+                error!(error = err, "couldn't start deployment");
+
+                DeploymentEvent::new(EventStatus::Error, err)
+                    .send(&id, &self.device)
+                    .await;
+
+                return;
+            }
         };
 
+        info!("stopping deployment");
+
         DeploymentEvent::new(EventStatus::Stopping, "")
-            .send(id.uuid(), &self.device)
+            .send(&id, &self.device)
             .await;
 
-        if let Err(err) = self.stop_node(node.id, node.idx).await {
+        if let Err(err) = self.stop_deployment(id, containers).await {
             let err = format!("{:#}", eyre::Report::new(err));
 
-            error!(error = err, "couldn't stop deployment");
+            error!(error = err, "couldn't start deployment");
```

Comment:
Is it a start or stop? From code seems a stop
```suggestion
            error!(error = err, "couldn't stop deployment");
```